src/config.rs

```diff
@@ -268,13 +296,10 @@ ZdJtrEnGRc0RGSBwP3N/fkcEAzuPw1ivcqOH5bWXPzSqPqQfADOrd8lK
 -----END PRIVATE KEY-----
 ";
 
-#[cfg(any(test))]
 const TEST_MATCHKEY_ENCRYPTION_KEY: &str = "\
```

Comment:
we use `test-fixture` feature extensively in CLI tools. It makes me very nervous that there exists a valid build for helper server that could reference those test keys. I am thinking if we accidentally build and deploy such configuration and someone realizes that IPA uses the test key then they can decrypt all match keys.

I like the setup Andy proposed in the TLS PR - have `test_setup` generate everything we need and put it inside `test_data` folder for default setup.  Generate `network.toml` that references those and pick it up during helper bootstrap. This may work well for HPKE configuration too

Reply:
I was able to hide it behind test-fixture feature flag for now. Some of plumbing for test setup should be done but I will complete it when I start coding the consumption in MPC.

Reply:
In #653 I have made some more changes to the way this test data is handled. `localhost_config` is gone, the `self-signed-certs` feature is gone, and some bits and pieces of code for generating test configs are consolidated into `src/net/test.rs`.

If this PR is ready soon, go ahead and merge it, and I'll resolve conflicts with #653, possibly moving some of the report key test data to match what I've done with the TLS test data.